sdk/synapse/azure-analytics-synapse-accesscontrol/src/main/java/com/azure/analytics/synapse/accesscontrol/AccessControlAsyncClient.java

```diff
@@ -0,0 +1,497 @@
+        return this.serviceClient.getRoleAssignmentsAsync(roleId, principalId, continuationToken);
+    }
+
+    /**
+     * List role assignments.
+     * 
+     * @param roleId Synapse Built-In Role Id.
+     * @param principalId Object ID of the AAD principal or security-group.
+     * @param continuationToken Continuation token.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return a list of role assignments.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<List<RoleAssignmentDetails>> getRoleAssignments(String roleId, String principalId, String continuationToken, Context context) {
+        return this.serviceClient.getRoleAssignmentsAsync(roleId, principalId, continuationToken, context);
+    }
+
+    /**
+     * List role assignments.
+     * 
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return a list of role assignments.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<List<RoleAssignmentDetails>> getRoleAssignments() {
+        return this.serviceClient.getRoleAssignmentsAsync();
+    }
+
+    /**
+     * Get role assignment by role assignment Id.
+     * 
+     * @param roleAssignmentId The ID of the role assignment.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return role assignment by role assignment Id.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SimpleResponse<RoleAssignmentDetails>> getRoleAssignmentByIdWithResponse(String roleAssignmentId) {
+        return this.serviceClient.getRoleAssignmentByIdWithResponseAsync(roleAssignmentId);
+    }
+
+    /**
+     * Get role assignment by role assignment Id.
+     * 
+     * @param roleAssignmentId The ID of the role assignment.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return role assignment by role assignment Id.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SimpleResponse<RoleAssignmentDetails>> getRoleAssignmentByIdWithResponse(String roleAssignmentId, Context context) {
```

Comment:
These should return `Response`, not `SimpleResponse`. Autorest was recently updated to return the correct type, so please update this API before GA.

Reply:
Now the method signature is as below
```java
    public Mono<Response<RoleAssignmentDetails>> getRoleAssignmentByIdWithResponse(String roleAssignmentId) 
```